src/state_types/models/context.rs

```diff
@@ -85,6 +102,12 @@ impl<Env: Environment + 'static> Update for Ctx<Env> {
                 self.content.addons.push(*descriptor.to_owned());
                 Effects::one(save_storage::<Env>(&self.content))
             }
+            Msg::Action(Action::Settings(ActionSettings::Store(settings))) => {
```

Comment:
not sure I like this - how about just passing the entire `settings` struct every time, so you would just `*self.content.settings = settings`